languages/elixir/concepts/pattern-matching/about.md

```diff
@@ -1 +1,52 @@
-TODO: add information on pattern-matching concept
+When writing Elixir functions, we can made use of an [assertive style][assertive-style] with [pattern matching][pattern-match-doc]:
+
+```elixir
+def read_file() do
+  {:ok, contents} = File.read("hello.txt")
+  contents
+end
+```
+
+- Pattern matching is explicitly performed using the match operator, [`=/2`][match-op].
+
+  - Matches succeed when the _shape_ of the data on the left side of the operator matches the right side
+  - When matches succeed, variables on the left are bound to the values on the right
+  - Using an underscore, `_`, allows us to disregard the values in those places
+
+    ```elixir
+    {:ok, number, _} = {:ok, 5, [4.5, 6.3]}
+
+    number
+    # => 5 is bound to this variable
+    ```
+
+  - [The pin operator `^`][getting-started-pin-operator] can be used to prevent rebounding a variable and instead pattern match against its existing value
```

Comment:
Now that the strategy changed and those documents will be accessible also outside of the context of the exercise, I wanted to make them more complete on their own, and the concept of pattern-matching is not complete without the pin operator.